logging/logger.go

```diff
@@ -3,40 +3,35 @@ package logging
 
 import (
 	"io"
-	"log"
+	"log/slog"
 	"os"
 )
 
-var logger *log.Logger
+const logFile = "revive.log"
+
+var logger *slog.Logger
 
 // GetLogger retrieves an instance of an application logger which outputs
 // to a file if the debug flag is enabled
-func GetLogger() (*log.Logger, error) {
+func GetLogger() (*slog.Logger, error) {
 	if logger != nil {
 		return logger, nil
 	}
 
-	var writer io.Writer
-	var err error
-
-	writer = io.Discard // by default, suppress all logging output
-	debugModeEnabled := os.Getenv("DEBUG") == "1"
-	if debugModeEnabled {
-		writer, err = os.Create("revive.log")
-		if err != nil {
-			return nil, err
-		}
+	debugModeEnabled := os.Getenv("DEBUG") != ""
+	if !debugModeEnabled {
+		// by default, suppress all logging output
+		return slog.New(slog.NewTextHandler(io.Discard, nil)), nil
```

Comment:
Please consider adding an inline comment about this

> For discarding output, I use slog.NewTextHandler(io.Discard, nil) because slog.DiscardHandler exist only in Go 1.24.

Right now, revive uses Go 1.23 in go.mod

Maybe an issue should be added to backlog about this.